backend/wmg/api/rollup.py

```diff
@@ -0,0 +1,137 @@
+import owlready2
+import pandas as pd
+import numpy as np
+import numba as nb
+from functools import lru_cache
+from backend.wmg.data.constants import CL_BASIC_PERMANENT_URL_OWL
+
+
+# ontology object
+onto = None
+
+
+def find_descendants_per_cell_type(cell_types):
+    """
+    Find descendants for each cell type in the input list.
+
+    Parameters
+    ----------
+    cell_types : list
+        List of cell types (cell type ontology term IDs) to find descendants for
+
+    Returns
+    -------
+    descendants_per_cell_type : list
+        List of lists of descendants for each cell type in the input list.
+    """
+
+    global onto
+    if not onto:
+        onto = owlready2.get_ontology(CL_BASIC_PERMANENT_URL_OWL)
```

Comment:
we might want to load this eagerly at server startup to avoid the load time on the first request that triggers this